src/main/java/org/javarosa/core/model/condition/EvaluationContext.java

```diff
@@ -323,8 +318,13 @@ public void setQueryContext(QueryContext queryContext) {
     private void expandReferenceAccumulator(TreeReference sourceRef, DataInstance sourceInstance,
                                             TreeReference workingRef, Vector<TreeReference> refs,
                                             boolean includeTemplates) {
-
-        int depth = workingRef.size();
+        int depth;
+        try {
+            depth = workingRef.size();
```

Comment:
From a style perspective, I'd generally do the most specific check possible (which is a direct check-and-throw), rather than a catch-and-throw, to prevent an issue in the future where more than one dereference occurs in the this block and it obfuscates the error, that also in theory prevents confusion if `workingRef.size()` throws an NPE internally rather than due to the dereference.

Reply:
Agree and made the change.